scripts/protein.py

```diff
@@ -0,0 +1,255 @@
+    :param current_encoding: specify current encoding (str)
+    :return: if seq is correct protein seq or not (bool)
+    """
+    temp_seq = seq.replace(' ', '')
+    if current_encoding == 'one':
+        for residue in temp_seq:
+            if residue.upper() not in RESIDUES_NAMES.values():
+                return False
+        return True
+    if current_encoding == 'three':
+        for residue_start_idx in range(0, len(temp_seq) - 2, 3):
+            residue = temp_seq[residue_start_idx:residue_start_idx + 3]
+            if residue.upper() not in RESIDUES_NAMES.keys():
+                return False
+        return True
+
+
+def get_seq_characteristic(seq: str) -> dict:
+    """
+    Count entry of each residue type in your seq. Get description of amino acid composition in dict format.
+    :param seq: protein seq in 1-letter encoding (str)
+    :return: each residue type in seq in 3-letter code and its amount in current seq (dict)
+    """
+    residue_count = {}
+    for residue in set(seq):
+        residue_entry = seq.count(residue)
+        residue_count[[three_letter_residue for three_letter_residue in RESIDUES_NAMES if
+                       RESIDUES_NAMES[three_letter_residue] == residue][0]] = residue_entry
+    return residue_count
+
+
+def find_residue_position(seq: str, res_of_interest: str) -> str:
+    """
+    Find all positions of certain residue in your seq
+    :param seq: protein seq in 1-letter encoding (str)
+    :param res_of_interest: specify the residue of interest in 1-letter encoding (str)
+    :return: positions of specified residue in your seq (str)
+    """
+    res_of_interest = res_of_interest.upper()
+    res_of_interest_positions = []
+    for ind, res in enumerate(seq, 1):
+        if res == res_of_interest:
+            res_of_interest_positions.append(ind)
+    return f"{res_of_interest} positions: {', '.join(map(str, res_of_interest_positions))}"
+
+
+def find_site(seq: str, site: str) -> str:
+    """
+    Find if seq contains certain site and get positions of its site
+    :param seq: protein seq in 1-letter encoding (str)
+    :param site: specify site of interest in 1-letter encoding (str)
+    :return: positions of residues for each certain site in seq (str)
+    """
+    site = site.upper()
+    if not is_protein(site, current_encoding='one'):
+        return f'Site {site} is not a protein!'
+    if site in seq:
+        site_full_position = []
+        site_count = seq.count(site)
+        site_start_position = [(coordinate + 1) for coordinate in range(len(seq)) if seq.startswith(site, coordinate)]
+        site_end_position = [(coordinate + len(site) - 1) for coordinate in site_start_position]
+        for idx in range(len(site_start_position)):
+            site_full_position.append(f'{site_start_position[idx]}:{site_end_position[idx]}')
+        return (f"Site entry in sequence = {site_count}. "
+                f"Site residues can be found at positions: {', '.join(site_full_position)}")
```

Comment:
Если ты хочешь сделать разрыв строки, чтобы строка сильно длинной не была, то лучше сделать так:
```suggestion
        f"Site entry in sequence = {site_count}. \
Site residues can be found at positions: {', '.join(site_full_position)}"
```